pkg/executor/scraper/filesystem_extractor.go

```diff
@@ -0,0 +1,55 @@
+package scraper
+
+import (
+	"context"
+	"os"
+
+	"github.com/kubeshop/testkube/pkg/filesystem"
+
+	"github.com/pkg/errors"
+)
+
+type FilesystemExtractor struct {
+	dirs []string
+	fs   filesystem.FileSystem
+}
+
+func NewFilesystemExtractor(dirs []string, fs filesystem.FileSystem) *FilesystemExtractor {
+	return &FilesystemExtractor{dirs: dirs, fs: fs}
+}
+
+func (e *FilesystemExtractor) Extract(ctx context.Context, process ProcessFn) error {
+	for _, dir := range e.dirs {
+		err := e.fs.Walk(
+			dir,
+			func(path string, fileInfo os.FileInfo, err error) error {
+				if err != nil {
+					return errors.Wrapf(err, "error walking path %s", path)
+				}
+
+				if fileInfo.IsDir() {
+					return nil
+				}
+
+				reader, err := e.fs.OpenFileBuffered(path)
+				if err != nil {
+					return errors.Wrapf(err, "error opening buffered %s", path)
+				}
+				object := &Object{
```

Comment:
Why not File - Object is quite generic name

Reply:
I wanted to build generic Extractor interface which should in theory be able to extract data from anywhere.
In our case, we only walk directories so we only have 1 implementation and we deal with files there.